pypesto/objective/base.py

```diff
@@ -422,17 +446,38 @@ def check_grad(
             fd_err_list.append(np.mean(fd_err_ix))
             abs_err_list.append(np.mean(abs_err_ix))
             rel_err_list.append(np.mean(rel_err_ix))
+            if detailed:
+                fval_p_list.append(fval_p)
+                fval_m_list.append(fval_m)
+                std_check_list.append(std_check_ix)
+                mean_check_list.append(mean_check_ix)
 
         # create dataframe
-        result = pd.DataFrame(data={
-            'grad': grad_list,
-            'fd_f': fd_f_list,
-            'fd_b': fd_b_list,
-            'fd_c': fd_c_list,
-            'fd_err': fd_err_list,
-            'abs_err': abs_err_list,
-            'rel_err': rel_err_list,
-        })
+        if detailed:
```

Comment:
create base dictionary and update if detailed?

Reply:
Ah nevermind, looks like `dict` also maintains order in recent Python versions...